/*
 * Copyright (C) 2025 The Android Open Source Project
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

syntax = "proto3";

package android.hardware.automotive.vehicle.proto;

/* Must be in sync with RawPropValues.aidl. */
message RawPropValues {
    /* This is used for properties of types VehiclePropertyType#INT
     * and VehiclePropertyType#INT_VEC */
    repeated int32 int32_values = 1;

    /* This is used for properties of types VehiclePropertyType#FLOAT
     * and VehiclePropertyType#FLOAT_VEC */
    repeated float float_values = 2;

    /* This is used for properties of type VehiclePropertyType#INT64 */
    repeated int64 int64_values = 3;

    /* This is used for properties of type VehiclePropertyType#BYTES */
    bytes byte_values = 4;

    /* This is used for properties of type VehiclePropertyType#STRING */
    string string_value = 5;
};
